pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -79,6 +86,11 @@ func (dsl *DaemonSetLock) Test(metadata interface{}) (holding bool, err error) {
 		if err := json.Unmarshal([]byte(valueString), &value); err != nil {
 			return false, err
 		}
+
+		if ttlExpired(value.Created, value.TTL) {
```

Comment:
What happens if you have a running kured install, and you add the TTL flag; won't it see the previous annotation as expired immediately?
Similarly if someone [takes the lock manually](https://github.com/weaveworks/kured#disabling-reboots)

Reply:
If you take lock manually as described, then set TTL flag it will invalidate it instantly as there will be no `created` and `ttl` values set in annotation:

`if ttl > 0 && time.Since(created) >= ttl`

Reply:
Do you think this is acceptable?

Reply:
So double checked this behaviour.

Launch params:
```
kured-pzs8b kured time="2020-05-05T18:27:40Z" level=info msg="Kubernetes Reboot Daemon: -64ebf53"
kured-pzs8b kured time="2020-05-05T18:27:40Z" level=info msg="Node ID: XXX.compute.internal"
kured-pzs8b kured time="2020-05-05T18:27:40Z" level=info msg="Lock Annotation: kube-system/kured:weave.works/kured-node-lock"
kured-pzs8b kured time="2020-05-05T18:27:40Z" level=info msg="Reboot Sentinel: /var/run/update_engine_autoupdate_completed every 1m0s"
kured-pzs8b kured time="2020-05-05T18:27:40Z" level=info msg="Blocking Pod Selectors: []"
kured-pzs8b kured time="2020-05-05T18:27:40Z" level=info msg="Reboot on: SunMonTueWedThuFriSat between 00:00 and 23:59 UTC"
kured-pzs8b kured time="2020-05-05T18:27:40Z" level=info msg="Force annotation cleanup after: 2m0s"
```

Manually disabled reboots:
```
➜  ~ kubectl -n kube-system annotate ds kured weave.works/kured-node-lock='{"nodeID":"manual"}'
daemonset.apps/kured annotated
```

Output:
```
kured-pzs8b kured time="2020-05-05T18:29:29Z" level=info msg="Reboot required"
kured-pzs8b kured time="2020-05-05T18:29:29Z" level=warning msg="Lock already held: manual"
kured-pzs8b kured time="2020-05-05T18:30:29Z" level=info msg="Reboot required"
kured-pzs8b kured time="2020-05-05T18:30:29Z" level=warning msg="Lock already held: manual"
kured-pzs8b kured time="2020-05-05T18:31:29Z" level=info msg="Reboot required"
kured-pzs8b kured time="2020-05-05T18:31:29Z" level=warning msg="Lock already held: manual"
kured-pzs8b kured time="2020-05-05T18:32:29Z" level=info msg="Reboot required"
kured-pzs8b kured time="2020-05-05T18:32:29Z" level=warning msg="Lock already held: manual"
kured-pzs8b kured time="2020-05-05T18:33:29Z" level=info msg="Reboot required"
kured-pzs8b kured time="2020-05-05T18:33:29Z" level=warning msg="Lock already held: manual"
```

Removing lock manually:
```
➜  ~ kubectl -n kube-system annotate ds kured weave.works/kured-node-lock-
daemonset.apps/kured annotated
```

Output:
```
kured-pzs8b kured time="2020-05-05T18:35:29Z" level=info msg="Reboot required"
kured-pzs8b kured time="2020-05-05T18:35:29Z" level=info msg="Acquired reboot lock"
kured-pzs8b kured time="2020-05-05T18:35:29Z" level=info msg="Draining node XXX.compute.internal"
...
```

So if lock has been taken manually, enabling this won't take over as I though initially.

To answer your question if this is acceptable - turns out an ability to manually disable reboot is not gone despite enabling annotation TTL functionality, so all is ok.

Of course this might not fit in every environments, but ideally I think the better solution would be to annotate node objects instead ds which would not require any cleanup mechanism being implemented.

Reply:
Thanks - I had missed that `if ttl > 0` is checking the ttl set by the locker (if any), not Kured's command-line setting.